### PR TITLE
Add impersonation option to e2e tests

### PR DESCRIPTION
**Pull request by AwesomePatrol**
It allows to impersonate a different service account to verify quota issues. See b/336784945#comment12 for more details.

Bug: b/336784945

### CODE REVIEW

File: e2e/testmain_test.go [hunk 1]
```diff
@@ -74,10 +80,25 @@ func TestMain(m *testing.M) {
...
 	ctx := context.Background()
-	client, err := google.DefaultClient(ctx, compute.ComputeScope)
+
+	credentials, err := google.FindDefaultCredentials(ctx, compute.ComputeScope)
 	if err != nil {
 		log.Fatal(err)
 	}
+	ts := credentials.TokenSource
+
+	if testFlags.serviceAccountName != "" {
```
Comment (kl52752) [inline e2e/testmain_test.go:90]: Can you add comment what credentials will be taken if this field is empty

File: e2e/testmain_test.go [hunk 2]
```diff
@@ -74,10 +80,25 @@ func TestMain(m *testing.M) {
...
+		ts, err = impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: testFlags.serviceAccountName,
+			Scopes:          []string{compute.ComputeScope, compute.CloudPlatformScope},
+			Lifetime:        1200 * time.Second,
+			Delegates:       []string{},
+		}, option.WithCredentials(credentials))
+		if err != nil {
+			log.Fatal(err)
```
Comment (kl52752) [inline e2e/testmain_test.go:98]: please extend the log with more info + testFlags.serviceAccountName

File: e2e/testmain_test.go [hunk 3]
```diff
@@ -74,10 +80,25 @@ func TestMain(m *testing.M) {
...
 	if err != nil {
 		log.Fatal(err)
 	}
+	ts := credentials.TokenSource
+
+	if testFlags.serviceAccountName != "" {
+		ts, err = impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: testFlags.serviceAccountName,
+			Scopes:          []string{compute.ComputeScope, compute.CloudPlatformScope},
+			Lifetime:        1200 * time.Second,
```
Comment (kl52752) [inline e2e/testmain_test.go:94]: 20minutes is this enough? Maybe we should parametrize this?

File: e2e/testmain_test.go (hunk 1 above)
Comment (AwesomePatrol) [reply e2e/testmain_test.go:90]: the behavior stays the same as before

File: e2e/testmain_test.go (hunk 3 above)
Comment (AwesomePatrol) [reply e2e/testmain_test.go:94]: it should refresh after this time elapses. I will verify tomorrow

Comment (AwesomePatrol) [reply e2e/testmain_test.go]: If `Lifetime` is not specified, it will auto-refresh (every 1 hour).

File: e2e/testmain_test.go (hunk 2 above)
Comment (AwesomePatrol) [reply e2e/testmain_test.go]: Done (although I believe that error messages in all functions called here are already sufficiently clear)

**Pull request merged**